modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -381,6 +401,43 @@ void visitBinarySpatialOperator(
         out.write(closingParenthesis);
     }
 
+    private void writeCurveIntersection(
+            BinarySpatialOperator filter, Expression e1, Expression e2, boolean swapped, Object extraData)
+            throws IOException {
+
+        out.write("ST_Distance(");
+        e1.accept(delegate, extraData);
+        out.write(", ");
+        e2.accept(delegate, extraData);
+        out.write(") ");
+
+        if (filter instanceof Disjoint) {
+            out.write("> 0");
+        } else {
+            out.write("= 0");
+        }
+    }
+
+    private boolean mayHaveCurves(Expression ex) {
+        if (ex instanceof Literal) {
+            Object value = ex.evaluate(null, Geometry.class);
+            return value instanceof CurvedGeometry
+                    || (value != null
+                            && (Geometry.class.equals(value.getClass())
```

Comment:
So the assumption is that generic Geometries/Collections may have curves?

Reply:
Indeed, generic geometrise and generic containers are not bound to a specific type, so they may contain curves.